### PR TITLE
macro refactor

### PR DESCRIPTION
**Pull request by tukanoidd**
Hey, here's the refactor I promised. :) 

It adds support for generics and also now allows you to provide any kinds of attributes separated by "|" inside of a literal string (I tried different approaches but could only make this one work without needing too much boilerplate) (can check in README).

Let me know what you think and if I need to change anything

### CODE REVIEW

**Comment by tukanoidd** [comment]
Also, I just realized that bc I'm using my work laptop, my work account was pushing the changes, sorry for the confusion if there was any regarding that :sweat_smile:

**Comment by tukanoidd** [comment]
oh, weird, a buid failure, ill be afk for a bit, and will check after, but thats weird, since it builds fine on my machine

**Comment by tukanoidd** [comment]
Oh, stable/nightly thing

File: iced_examples/counter/src/main.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ enum WindowDirection {
     Bottom,
 }
 
-#[to_layer_message(derives = "Debug Clone")]
+#[to_layer_message(attrs = "derive(Debug, Clone)")]
```
Comment (Decodetalkers) [inline iced_examples/counter/src/main.rs:41]: I want this be 

```rust
#[derive(Debug, Cone)]
```

**Comment by Decodetalkers** [comment]
Ok , I made some adjustments

**Comment by Decodetalkers** [comment]
Ok, now I think it is fine

**Comment by tukanoidd** [comment]
Hmm, I guess it's fair, for some reason I completely forgot the fact I could just take the attrs from DeriveInput (mb bc I was so hung up on the fact that it was panicing if I tried adding attrs to MessageEnum :sweat_smile: )

**Comment by Decodetalkers** [comment]
Do you mind adding similar macro to iced-sessionlock?

**Comment by tukanoidd** [comment]
I personally havent used iced-sessionlock so I'm going to be not familiar with the API but I could take a look after work (in about 6-7 hrs) :)

**Comment by Shinyzenith** [comment]
Hey @tukanoidd thank you so much for your high quality contributions. As you seem to be using @Decodetalkers library in your projects, would be nice to have your say on our changes.

May I invite you to our GitHub org?

**Comment by tukanoidd** [comment]
> Hey @tukanoidd, thank you so much for your high-quality contributions. As you seem to be using the @Decodetalkers library in your projects, it would be nice to have your say on our changes.
> 
> May I invite you to our GitHub org?

I mean, if you don't mind, sure, although I really don't know how much help I am going to be when it comes to actual logic implementation, as I'm still quite unskilled in this area.

**Pull request merged**